r4/src/main/java/org/opencds/cqf/r4/providers/MeasureOperationsProvider.java

```diff
@@ -222,41 +222,94 @@ public MeasureReport evaluateMeasure(@IdParam IdType theId,
     // }
 
     @Operation(name = "$care-gaps", idempotent = true, type = Measure.class)
-    public Parameters careGapsReport(@OperationParam(name = "periodStart") String periodStart,
-            @OperationParam(name = "periodEnd") String periodEnd, @OperationParam(name = "subject") String subject,
-            @OperationParam(name = "topic") String topic, @OperationParam(name = "practitioner") String practitioner,
-            @OperationParam(name = "measureId") List<String> measureId, @OperationParam(name = "measureIdentifier") List<String> measureIdentifier,
-            @OperationParam(name = "measureUrl") List<CanonicalType> measureUrl, @OperationParam(name="status")List<String> status,
-            @OperationParam(name = "organization") String organization,  @OperationParam(name = "program") String program){
+    public Parameters careGapsReport(
+        @OperationParam(name = "periodStart") List <String> periodStart,
+        @OperationParam(name = "periodEnd") List <String> periodEnd,
+        @OperationParam(name = "subject") List <String> subject,
+        @OperationParam(name = "topic") String topic,
+        @OperationParam(name = "practitioner") String practitioner,
+        @OperationParam(name = "measureId") List <String> measureId,
+        @OperationParam(name = "measureIdentifier") List <String> measureIdentifier,
+        @OperationParam(name = "measureUrl") List <CanonicalType > measureUrl,
+        @OperationParam(name = "status") List <String> status,
+        @OperationParam(name = "organization") String organization,
+        @OperationParam(name = "program") String program) {
         //TODO: topic should allow many and be a union of them
         //TODO: "The Server needs to make sure that practitioner is authorized to get the gaps in care report for and know what measures the practitioner are eligible or qualified."
         Parameters returnParams = new Parameters();
-        
-        if (careGapParameterValidation(periodStart, periodEnd, subject, topic, practitioner, measureId, measureIdentifier, measureUrl, status, organization, program)) {
-            List<Measure> measures = resolveMeasures(measureId, measureIdentifier, measureUrl);
-            if (subject.startsWith("Patient/")){
-                resolvePatientGapBundleForMeasures(periodStart, periodEnd, subject, topic, status, returnParams, measures, "return", organization);
-            } else if (subject.startsWith("Group/")) {
-                returnParams.setId((status==null?"all-gaps": status) + "-" + subject.replace("/","_") + "-report");
-                (getPatientListFromGroup(subject))
-                    .forEach(groupSubject -> resolvePatientGapBundleForMeasures(periodStart, periodEnd, subject, topic, status, returnParams, measures, "return", organization));
-            }
-            else if (Strings.isNullOrEmpty(practitioner)) {
+
+        // Setting periodStart, periodEnd, and subject to lists to check if multiple have been supplied.
```

Comment:
Could we move this all in careGapsParameterValidation as well?

This all looks proximate and should be continued within the same method.